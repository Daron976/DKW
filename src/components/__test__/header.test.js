import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Calculator from '../calculator';
import Header from '../header';
import Home from '../home';
import Quote from '../quote';

describe('Header component', () => {
  const Route = () => {
    <BrowserRouter>
      <Header />
    </BrowserRouter>;
  };

  test('header renders as exepcted', async () => {
    const snapHeader = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
      .toJSON();

    expect(snapHeader).toMatchSnapshot();
  });

  test('Header is present in the Dom', async () => {
    render(<Route />);

    const headerComponent = screen.findByRole('heading');

    expect(headerComponent).toBeInTheDocument;//eslint-disable-line
  });

  test('Header is visible', async () => {
    render(<Route />);

    const headerComponent = screen.findByRole('heading');

    expect(headerComponent).toBeVisible;//eslint-disable-line
  });

  describe('navigation', () => {
    test('links should be present', async () => {
      render(<Route />);

      const homeLink = screen.findByRole('link', { name: Home });
      const calculatorLink = screen.findByRole('link', { name: Calculator });
      const quoteLink = screen.findByRole('link', { name: Quote });

      expect(homeLink).toBeInTheDocument;//eslint-disable-line
      expect(calculatorLink).toBeInTheDocument;//eslint-disable-line
      expect(quoteLink).toBeInTheDocument;//eslint-disable-line
    });

    test('links should be visible', async () => {
      render(<Route />);

      const homeLink = screen.findByRole('link', { name: Home });
      const calculatorLink = screen.findByRole('link', { name: Calculator });
      const quoteLink = screen.findByRole('link', { name: Quote });

      expect(homeLink).toBeVisible;//eslint-disable-line
      expect(calculatorLink).toBeVisible;//eslint-disable-line
      expect(quoteLink).toBeVisible;//eslint-disable-line
    });
  });
});
