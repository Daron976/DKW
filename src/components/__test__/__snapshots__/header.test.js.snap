// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component header renders as exepcted 1`] = `
<header
  className="header"
>
  <p
    className="logo"
  >
    <strong>
      Math Magician
    </strong>
  </p>
  <nav>
    <ul
      className="nav-link"
    >
      <li
        className="link-item"
        data-testid="home"
        id="home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="link-item"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="link-item"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
